### PR TITLE
Update download_models.py

### PR DESCRIPTION
**Pull request by sturkmen72**
Update download_models.py (actual hashes)

### CODE REVIEW

**Comment by sturkmen72** [comment]
update is according to actual hashes after downloading models

see 

```
Model <CNN Mnist (ONNX)>
  expect ce4fb4914cd1d9e0faed3294e5cecfd1847339763
  actual e4fb4914cd1d9e0faed3294e5cecfd1847339763
  hash check failed - extracting
  get mnist/model.onnx
  progress > done
```

**Comment by dkurt** [comment]
@sturkmen72, They did it again :confused: . Could you please update the branch using my one: https://github.com/dkurt/opencv_extra/tree/Update_download_models_py?

File: testdata/dnn/download_models.py [hunk 1]
```diff
@@ -673,25 +679,25 @@ def save(self, r):
...
         member='emotion_ferplus/model.onnx',
-        sha='3c0546c99910156aed7d7809fe7be9bd92168b05',
+        sha='2ef5b3a6404a5feb8cc396d66c86838c4c750a7e',
         filename='onnx/models/emotion_ferplus.onnx'),
     Model(
         name='Emotion FERPlus (ONNX)',
         archive='emotion_ferplus.tar.gz',
         member='emotion_ferplus/test_data_set_0/input_0.pb',
-        sha='3a3189093cc38a6ffab10c65eda90a24e12d7e1a',
+        sha='29621536528116fc12f02bc81c7265f7ffe7c8bb',
```
Comment (sturkmen72) [inline testdata/dnn/download_models.py]: @dkurt ready with additional change ( line 694 )

**Comment by mshabunin** [comment]
My branch also have different paths and hashes for ResNet101_DUC_HDC: https://github.com/opencv/opencv_extra/compare/master...mshabunin:fix-download-models
:wink:

File: testdata/dnn/download_models.py [hunk 2]
```diff
@@ -816,3 +822,9 @@ def save(self, r):
     for m in models:
         print(m)
         m.get()
```
Comment (mshabunin) [inline testdata/dnn/download_models.py]: BTW, we can slightly simplify final verification:
```.python
def get(self):
    ...
    return self.verify()
...
failedSums = []
for m in models:
        print(m)
        if not m.get():
            failedSums += m.filename
if failedSums:
    print('Checksum verification failed for following files: {}'.format(failedSums))
    exit(1)
```

Comment (sturkmen72) [reply testdata/dnn/download_models.py]: @mshabunin could you add necessary changes please.

Comment (mshabunin) [reply testdata/dnn/download_models.py]: @sturkmen72 , I've updated my branch, you can pull it and verify in your environment.

Comment (sturkmen72) [reply testdata/dnn/download_models.py]: @mshabunin thank you. i tested and believe it is ready!

**Comment by alalek** [comment]
This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into 3.4 branch first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

**Pull request merged**